core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2319,22 +2319,27 @@ public boolean isRootUrlSecure() {
         String scheme = getXForwardedHeader(req, "X-Forwarded-Proto", req.getScheme());
         buf.append(scheme).append("://");
         String host = getXForwardedHeader(req, "X-Forwarded-Host", req.getServerName());
-        int index = host.indexOf(':');
+        int index = host.lastIndexOf(':');
```

Comment:
any chance it changes the behavior for IPv4.
`buf.append(host, 0, index)` may start behaving differently if there are two semicolons in the hostname. Not sure it is possible.

Reply:
I don't think it would change that behavior. Not all hostnames abide by the RFCs, but the scenario you describe doesn't seem possible or reasonable.